### PR TITLE
extension and uppercase support

### PR DESCRIPTION
**Pull request by JaSei**

### CODE REVIEW

**Review by augi — DISMISSED** [review]
I would prefer to have a default value (`false`) for `upper` parameter.

**Comment by tciml** [comment]
👍

File: main.go [hunk 1]
```diff
@@ -51,6 +51,8 @@ var (
 	logJson       = kingpin.Flag("json", "log in json format").Bool()
 	retryDelay    = kingpin.Flag("delay", "exponential retry - start delay time").Default(storclient.DefaultRetryDelay.String()).Duration()
 	retryAttempts = kingpin.Flag("attempts", "count of attempts of retry").Default(strconv.Itoa(storclient.DefaultRetryAttempts)).Uint()
+	extension     = kingpin.Flag("extension", "downloaded file extension - like '.dat' = SHA.dat").Default("").String()
```
Comment (ondrej-hosak) [inline main.go]: What about some check for '.' in extension? Otherwise I would call that variable suffix.

File: client/download.go [hunk 2]
```diff
@@ -54,7 +54,16 @@ func (client *StorClient) downloadWorker(id int, httpClient httpClient, shasForD
 			return
 		}
 
-		filepath, err := pathutil.NewPath(client.downloadDir, sha.String())
+		filename := sha.String()
+		if client.UpperCase {
+			filename = strings.ToUpper(sha.String())
+		}
+
+		if client.Extension != "" {
```
Comment (ondrej-hosak) [inline client/download.go]: I think this if is redundant

Comment (JaSei) [reply client/download.go]: good point, I fix it...

File: main.go (hunk 1 above)
Comment (JaSei) [reply main.go]: suffix is better name - thanks - I fix it..

**Review by JaSei — COMMENTED** [review]
@augi default of `upper` option is `false`... (Default of all `bool` variables in go is `false`)

**Pull request merged**